dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/DLCOracle.scala

```diff
@@ -193,31 +267,60 @@ case class DLCOracle(private val extPrivateKey: ExtPrivateKeyHardened)(implicit
               s"No event saved with nonce ${nonce.hex} $outcome"))
       }
 
-      eventOutcomeOpt <- eventOutcomeDAO.read((nonce, outcome))
-      eventOutcomeDb <- eventOutcomeOpt match {
-        case Some(value) => Future.successful(value)
-        case None =>
-          Future.failed(new RuntimeException(
-            s"No event outcome saved with nonce and message ${nonce.hex} $outcome"))
+      (outcomeBytes, newDb) <- eventDb.eventDescriptorTLV match {
+        case enumV0: EnumEventDescriptorV0TLV =>
+          require(enumV0.outcomes.contains(outcome),
+                  "Outcome given is not a possible outcome for this event")
+          Future.successful((CryptoUtil.serializeForHash(outcome), eventDb))
+        case rangeV0: RangeEventDescriptorV0TLV =>
+          val int32 = Int32(outcome.toLong)
+          require(rangeV0.outcomes.contains(int32),
+                  "Outcome given is not a possible outcome for this event")
+          Future.successful((int32.bytes, eventDb))
+        case external: ExternalEventDescriptorV0TLV =>
+          if (external == oldDbRepDescriptor) {
```

Comment:
So is there a reason we are doing this here rather than just doing something on server startup similar to what we did we `runChainWorkRecalc()`? Then we don't have to worry about keeping this logic in here forever 

https://github.com/bitcoin-s/bitcoin-s/blob/30b6e22662258baac1b802c3e263760a66a8542d/app/server/src/main/scala/org/bitcoins/server/BitcoinSServerMain.scala#L238-L237

Reply:
In my mind, all this legacy migration stuff would happen inside of `DLCOracleAppConfig.start()` so a user can just start the module, we will apply migrations and also migrate legacy data if any exists.

Reply:
Looking at this even more, unfortunately it looks like we screwed up our chainwork migration. If a user was using bitcoin-s chain module pre-chainwork, and then updated their binary, the `runChainWorkRecalc()`would not get run as it's not affiliated with the chain module, but just run in the `BitcoinSServerMain`class. With breaking changes like this we need to make sure they are affiliated with the specific module rather than the stand alone binary

https://github.com/bitcoin-s/bitcoin-s/blob/30b6e22662258baac1b802c3e263760a66a8542d/app/server/src/main/scala/org/bitcoins/server/BitcoinSServerMain.scala#L59-L58

Reply:
Waiting on #2173